substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/thread/VMThreadMTFeature.java

```diff
@@ -190,7 +192,16 @@ private static ValueNode currentThread(GraphBuilderContext b, boolean usedForAdd
          * must not directly reference a fixed register).
          */
         boolean isDeoptTarget = b.getMethod() instanceof SharedMethod && ((SharedMethod) b.getMethod()).isDeoptTarget();
-        if (isDeoptTarget || usedForAddress) {
+
+        /*
+         * Due to the fact that the LLVM backend handles reading the thread pointer in entry point
+         * methods as a stack slot load instead of a direct register access, a ReadRegisterFixedNode
+         * should be emitted in those cases so that the register read doesn't get hoisted above
+         * where the thread pointer gets stored in the stack slot.
+         */
+        boolean isLLVM = SubstrateOptions.CompilerBackend.getValue().equals("llvm");
```

Comment:
I would just remove `isLLVM` here. It does not hurt to fix those registers on both backends.

Reply:
The emission of floating nodes was explicitly designed for efficiency, but I guess this isn't too frequent.